File: src/main/scala/s/chain_of_resposibility/inheritance/supports/SpecialSupport.scala
package s.chain_of_resposibility.inheritance.supports

import s.chain_of_resposibility.Trouble
import s.chain_of_resposibility.inheritance.Support

final class SpecialSupport(name: String, private val number: Int, next: Support)
    extends Support(name, next) {
  def this(name: String, number: Int) {
    this(name, number, null)
  }

  override protected def resolve(trouble: Trouble): Boolean =
    trouble.getNumber == number
}

File: src/main/java/j/chain_of_responsibility/inheritance/supports/SpecialSupport.java
package j.chain_of_responsibility.inheritance.supports;

import j.chain_of_responsibility.Trouble;
import j.chain_of_responsibility.inheritance.Support;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class SpecialSupport extends Support {
  private final int number;

  public SpecialSupport(@NotNull String name, int number, @Nullable Support next) {
    super(name, next);
    this.number = number;
  }

  SpecialSupport(@NotNull String name, int number) {
    this(name, number, null);
  }

  @Override
  protected boolean resolve(Trouble trouble) {
    return trouble.getNumber() == number;
  }
}
